### PR TITLE
Support for v1.29

### PR DESCRIPTION
**Pull request by mysticaltech**
Made possible by #1231 and a fix to the autoscaler by @Silvest89  🚀

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -454,7 +454,7 @@ variable "enable_metrics_server" {
 
 variable "initial_k3s_channel" {
   type        = string
-  default     = "v1.28" # Please update kube.tf.example too when changing this variable
+  default     = "v1.29" # Please update kube.tf.example too when changing this variable
```
Comment (aleksasiriski) [inline variables.tf:457]: Please update kube.tf.example too when changing this variable

**Review by aleksasiriski — APPROVED** [review]

**Comment by mysticaltech** [comment]
@Silvest89 Could you please take care of updating the autoscaler mother repo so that our fork does not diverge 🙏 Hopefully soon enough we can go back to using that and not have to maintain our own.

**Comment by M4t7e** [comment]
Hey @mysticaltech, I want to make sure we maintain compatibility with our 3rd party packages. The developers behind these packages are diligently working to keep up with new Kubernetes releases, which often means addressing incompatibilities/bugs or implementing new features.

I collected the relevant minimum versions:
| K3s  | Autoscaler | Calico | Cilium | HCCM | HCSI | Kured |
|------|------------|--------|--------|------|------|-------|
| 1.26 | 1.26       | 3.25   | 1.13   | 1.14 | 2.2  | 1.12  |
| 1.27 | 1.27       | 3.26   | 1.14   | 1.16 | 2.4  | 1.13  |
| 1.28 | 1.28       | 3.25   | 1.15   | 1.18 | 2.5  | 1.14  |
| 1.29 | 1.29       | 3.27   | 1.15   | 1.19 | 2.6  | 1.15  |

Versions after this PR:
- **Autoscaler:** 20240226 -> 1.29 (OK)
- **Calico:** 3.26.4 (Not OK)
- **Cilium:** 1.15.1 (OK)
- **HCCM:** latest (OK)
- **HCSI:** latest (OK)
- **Kured:** latest (OK)

It seems only Calico needs an upgrade. Fortunately, version 3.27 is already available.
I'm a bit confused about our Autoscaler versioning, but I understood this version reflects 1.29.

For the packages where we currently use the latest version, I would propose (maybe at some point in future) to adopt a more stringent versioning strategy to guarantee compatibility without being the first ones hitting day 1 issues of fresh releases. I prefer stability over being the early adopter (or software tester ^^). Over time, we might also face backward compatibility issues, for instance, if a package makes a major upgrade that requires a strict minimum Kubernetes version. Currently, our setup appears to install it without considering these constraints.

**Comment by mysticaltech** [comment]
@M4t7e  Thanks for the info, super important indeed! Good to hear that Calico 3.27 is available, I will bump it. 

Definitely, same here I prefer we use stable, but this issue #1231 initially led me to believe that we needed to move to v1.19 to fix the autoscaler issue. I later learned that autoscaler v1.29.x is in fact compatible with previous versions of kube too. But what's done is done.

About the autoscaler thing, what happened is that @Silvest89 contributed vital fixes and support for labelling Hetzner nodes if I remember correctly, it was merged on the mother repo kubernetes/autoscaler, but their release schedule is super long, so it just shipped a few weeks ago, in the meantime we had to fork our own (@Silvest89 did it) for users to benefit from a working version with Hetzner. Now same as before, a fix needs to be synced to the mother repo and it will take a long time to be released, so we are stuck with our own fork for a while longer. And yes, we always have the master of our fork updated from upstream via a bot.

**Comment by Silvest89** [comment]
@mysticaltech I've already opened a PR

**Comment by mysticaltech** [comment]
Thank you @Silvest89, really appreciate it man 🙏 Please keep up posted when merged  🤞

**Pull request merged**